sdk/assertion_provider.go

```diff
@@ -0,0 +1,29 @@
+package sdk
+
+import (
+	"context"
+)
+
+const (
+	// SchemaWildcard is a wildcard pattern that matches any assertion schema.
+	SchemaWildcard = "*"
+)
+
+type AssertionBinder interface {
+	// Bind creates and signs an assertion, binding it to the given manifest.
+	// The implementation is responsible for both configuring the assertion and binding it.
+	Bind(ctx context.Context, m Manifest) (Assertion, error)
```

Comment:
we only bind to the aggregate root signature hash today. do we want to expose the full manifest here?

Reply:
the idea is to simplify the Bind to known values, and not conditional useHex, schema version checks to get the aggregate hash, use the signed hash in the manifest

Reply:
we need `m Manifest` to determine `useHex` too.  so it is needed